google/search/tool.gpt

```diff
@@ -6,12 +6,12 @@ Share Tools: Search
 
 ---
 Name: Search
-Description: Search Google with a given query and return relevant information from the search results
+Description: Search Google with a given query and return relevant information from the search results. Search with more maxResults if you need more information.
 JSON Response: true
 Share Context: ../../time
 Tools: service
-Args: maxResults: The maximum number of search results to gather relevant information from (optional, default 3)
 Args: query: A question, statement, or topic to search with (required)
+Args: maxResults: The maximum number of search results to gather relevant information from (optional, default 3, minimum 2)
```

Comment:
The LLM isn't always great at respecting tool parameter descriptions. I'd also add some validation on the code side and return an error describing the lower (and potentially upper) bounds here.

Reply:
I agree with your point about validation, but I think it’s more appropriate to address this on the `gptscript` tool parameter side rather than in this PR. Specifically, we should introduce support for the `required` field for function definitions and `minimum`/`maximum` fields for numeric properties like this. For example:

```json
{
  "name": "get_n_day_weather_forecast",
  "description": "Get an N-day weather forecast",
  "strict": false,
  "parameters": {
    "type": "object",
    "properties": {
      "location": {
        "type": "string",
        "description": "The city and state, e.g., San Francisco, CA"
      },
      "format": {
        "type": "string",
        "enum": ["celsius", "fahrenheit"],
        "description": "The temperature unit to use. Infer this from the user's location."
      },
      "num_days": {
        "type": "integer",
        "description": "The number of days to forecast",
        "minimum": 3,
        "maximum": 10
      }
    },
    "required": ["location", "format", "num_days"]
  }
}
```

This kind of structure ensures validation happens at the tool definition level, keeping function parameters clean and easy to enforce boundaries. I’d be happy to explore this further in a separate PR or discussion thread!

Reply:
approving, we can follow up with code-level validation if it becomes an issue.